tqdm/tests/tests_tqdm.py

```diff
@@ -116,8 +116,11 @@ def write(self, s):
         self.text += s
         self.cursor = len(self.text)
 
-    def read(self):
-        return self.text[self.cursor:]
+    def read(self, n=None):
+        _cur = self.cursor
+        self.cursor = len(self.text) if n is None \
+            else min(_cur + n, len(self.text))
+        return self.text[_cur:self.cursor]
```

Comment:
A bit convoluted to use the same variable self.cursor, but ok, I see what you wanted to do. Maybe you should set default `n=-1` to be consistent with [Python io lib](https://docs.python.org/2/library/io.html#io.RawIOBase.read)?